polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1425,6 +1452,57 @@ pub mod pallet {
 				weight_limit,
 			)
 		}
+
+		/// Authorize another `aliaser` location to alias into the local `origin` making this call.
+		///
+		/// Usually useful to allow your local account to be aliased into from a remote location
+		/// also under your control (like your account on another chain).
+		///
+		/// WARNING: make sure the caller `origin` (you) trusts the `aliaser` location to act in
+		/// their/your name. Once authorized using this call, the `aliaser` can freely impersonate
+		/// `origin` in XCM programs executed on the local chain.
+		#[pallet::call_index(14)]
+		pub fn add_authorized_alias(
```

Comment:
Should we allow this indefinitely or always require an expiry? Thinking of the case where you do trust your account on another chain but maybe that chain gets compromised and you forgot to remove your authorization. Then the parachain can impersonate your account on their chain and in turn impersonate your account on this chain.

An expiry would be annoying but would make it safer. Apps could batch this with the actual action you want to take if they know the expiry is past.

Reply:
I see expiry as a way to not allow users to forget that they allowed some location to impersonate them. I think it's a UX improvement to allow users to forget but still clean up after themselves.

The reason why I'm proposing this is because it's not only the location they explicitly allow that could be the problem, they open the surface area of bugs to bugs on the other chain.

I'd expect the usage pattern to be B, but if it is A we can still make the expiry refresh every time they make an action with it.

On the implementation side I don't see how it's so complex and expensive. You'd just keep the block number timeout and disallow aliases that are past that.

Reply:
added expiry here https://github.com/paritytech/polkadot-sdk/pull/6336/commits/a606607344f688aeed2912bffbf233654cadc6f0, let me know what you think